File: backend-ms1/src/main/java/ma/maneo/khola/service/facade/admin/site/SiteImageAdminService.java
package ma.maneo.khola.service.facade.admin.site;

import java.util.List;
import ma.maneo.khola.bean.core.site.SiteImage;
import ma.maneo.khola.dao.criteria.core.site.SiteImageCriteria;
import ma.maneo.khola.zynerator.service.IService;



public interface SiteImageAdminService extends  IService<SiteImage,SiteImageCriteria>  {

    List<SiteImage> findBySiteId(Long id);
    int deleteBySiteId(Long id);
    long countBySiteG2r(String g2r);



}

File: frontend/src/app/controller/service/admin/site/SiteImageAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {SiteImageDto} from 'src/app/controller/model/site/SiteImage.model';
import {SiteImageCriteria} from 'src/app/controller/criteria/site/SiteImageCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class SiteImageAdminService extends AbstractService<SiteImageDto, SiteImageCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/siteImage/');
    }

    public constrcutDto(): SiteImageDto {
        return new SiteImageDto();
    }

    public constrcutCriteria(): SiteImageCriteria {
        return new SiteImageCriteria();
    }
}
